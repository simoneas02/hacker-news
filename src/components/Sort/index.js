import React from "react";
import classNames from "classnames";

import Button from "../Button";

const Sort = ({ sortKey, activeSortKey, onSort, children }) => {
  const sortClass = classNames("button-inline", {
    "button-active": sortKey === activeSortKey
  });

  return (
    <Button onClick={() => onSort(sortKey)} className={sortClass}>
      {children}
    </Button>
  );
};

export default Sort;
