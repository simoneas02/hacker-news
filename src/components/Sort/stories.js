import React from 'react';
import { storiesOf } from '@storybook/react';

import Sort from '.';

storiesOf('Sort', module).add("I'm Sort", () => <h1>Hello Sort</h1>);
