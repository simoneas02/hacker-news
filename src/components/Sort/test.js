import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Sort from ".";

describe("Sort", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Sort />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test("has a valid snapshot", () => {
    const component = renderer.create(<Sort />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
