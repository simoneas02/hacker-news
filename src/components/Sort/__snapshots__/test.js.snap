// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort has a valid snapshot 1`] = `
<button
  className="button-inline button-active"
  onClick={[Function]}
  type="button"
/>
`;
